gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java

```diff
@@ -104,7 +79,23 @@ public RequestT read(JsonReader in) {
   public Map<String, List<String>> getQueryParamNames(RequestT apiMessage) {
     Set<String> paramNames = getQueryParamNames();
     Map<String, List<String>> queryParams = new HashMap<>();
-    Map<String, List<String>> nullableParams = apiMessage.populateFieldsInMap(paramNames);
+    Map<String, List<String>> nullableParams = new HashMap<>();
+    for (String paramName : paramNames) {
+      Object paramValue = apiMessage.getFieldValue(paramName);
+      List<String> valueList;
+      if (paramValue == null) {
+        continue;
+      }
+      if (paramValue instanceof List) {
+        valueList = new LinkedList<>();
```

Comment:
I think we should prefer ArrayList over LinkedList in general. Unless you believe LinkedList will work faster?

See also: https://www.youtube.com/watch?v=YQs6IC-vgmo